src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -293,15 +295,23 @@ public void InitializeService(ServiceHost serviceHost, SqlToolsContext context)
             }
         }
 
-// turn off this code until needed (10/28/2016)
-#if false
-        private static async Task HandleDefinitionRequest(
-            TextDocumentPosition textDocumentPosition,
-            RequestContext<Location[]> requestContext)
+        private async Task HandleDefinitionRequest(TextDocumentPosition textDocumentPosition, RequestContext<Location[]> requestContext)
         {
-            await Task.FromResult(true);
+            //TODO: check for settings
+
+            Location[] locations = this.GetDefinition(textDocumentPosition);
+            if (locations != null)
+            {
+                await requestContext.SendResult(locations);
+            }
+            else
+            {
+                await Task.FromResult(true);
```

Comment:
I believe this else branch isn't necessary. As long as there's at least one await in the async method, there's no need to add a separate `Task.FromResult(true)`